DogsList/DogsList/Controllers/ListDogViewController.swift

```diff
@@ -0,0 +1,186 @@
+            
+            switch indexPath.section {
+            case 0:
+                guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "TopCell", for: indexPath) as? TopCell else { fatalError("Cannot create the cell") }
+                cell.congigure(urlString: imageURL)
+                return cell
+            case 1:
+                guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "MiddleCell", for: indexPath) as? MiddleCell else { fatalError("Cannot create the cell") }
+                cell.congigure(urlString: imageURL)
+                return cell
+            default:
+                guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "BottomCell", for: indexPath) as? BottomCell else { fatalError("Cannot create the cell") }
+                cell.congigure(urlString: imageURL)
+                return cell
+            }
+        })
+    }
+    
+    func add(imageURL: [String], animate: Bool = true) {
+        guard let dataSource = self.dataSource else { return }
+        DispatchQueue.main.async {
+            self.snapshot.appendSections([.first, .second, .third])
+            self.snapshot.appendItems([imageURL[0]], toSection: .first)
+            self.snapshot.appendItems([imageURL[1], imageURL[2]], toSection: .second)
+            self.snapshot.appendItems([imageURL[3], imageURL[4], imageURL[5]], toSection: .third)
+            dataSource.apply(self.snapshot, animatingDifferences: false)
+        }
+    }
+    
+    func alertGeenrate(alertTitle: String, alertMessage: String, actionTitle: String, handler: ((UIAlertAction) -> Void)? = nil) {
+        let alert = UIAlertController(title: alertTitle, message: alertMessage, preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: actionTitle, style: .default) {(_: UIAlertAction!) in
+            self.viewDidLoad()})
+        present(alert, animated: true, completion: nil)
+    }
+
+}
+
+// MARK: - DogManagerDelegate
+
+extension ListDogViewController: DogManagerDelegate {
+    func addDogs(dogImages: [String]) {
+        self.add(imageURL: dogImages)
+    }
+    
+    func startActivityIndicator() {
+        indicator.center = self.view.center
+        indicator.color = #colorLiteral(red: 1, green: 1, blue: 1, alpha: 1)
+        view.backgroundColor = #colorLiteral(red: 0.6967710582, green: 0.4893499848, blue: 0.1820901042, alpha: 1)
+        indicator.startAnimating()
+        view.addSubview(indicator)
+        collectionView.isHidden = true
+    }
+    
+    func stopActivityIndicator() {
+        DispatchQueue.main.async {
+            self.indicator.hidesWhenStopped = true
+            self.indicator.stopAnimating()
+            self.collectionView.isHidden = false
+        }
+    }
+    
+    func didFailWithError(error: Error) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
```

Comment:
What is the delay for?

Reply:
Thanks. The delay allows the alert does not overlap the activity indicator.